### PR TITLE
Adds zendesk request function, post request when correction is requested

### LINKED ISSUE #2114: Integrate revision requests into Zendesk workflow
we would like to track edit requests via zendesk.

acceptance criteria:
- integrate revision requests with zendesk
- a user should be able to submit a revision request and we receive a request in zendesk.
- CC the original submitter

### PR DESCRIPTION
**Pull request by dakota002**
Resolves #2114

### CODE REVIEW

**Comment by dakota002** [comment]
@lpsinger should I include a refactor of the contact form zendesk request in this PR?

**Review by lpsinger — COMMENTED** [review]
Is there an existing TypeScript library for the Zendesk API?

**Comment by dakota002** [comment]
Their website lists 2 nodejs packages. Neither are officially theirs, and one is not maintained anymore. 

This is the one that is currently maintained: https://github.com/blakmatrix/node-zendesk 

Might be worth investigating, but with our current usage it seems a bit bulky. It could be a worthwhile addition in the future though if we come up with some other admin features that want to make use of/interact with  it

File: app/lib/zendesk.server.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+interface Requester {
+  email: string
+  name: string
+}
+
+interface RequestComment {
+  body: string
+}
+
+export async function postToZendesk(request: ZendeskRequest) {
```
Comment (lpsinger) [inline app/lib/zendesk.server.ts:26]: Suggest more descriptive function name:

```suggestion
export async function postZendeskRequest(request: ZendeskRequest) {
```

File: app/lib/zendesk.server.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        request,
+      }),
+    }
+  )
+
+  if (!response.ok) {
+    console.error(response)
+    throw new Error(`Reqeust failed with status ${response.status}`)
+  }
+}
```
Comment (lpsinger) [inline app/lib/zendesk.server.ts:48]: This code is now duplicated in contact.tsx. Please refactor to eliminate multiple copies.

**Review by lpsinger — CHANGES_REQUESTED** [review]
There is now d

**Review by lpsinger — CHANGES_REQUESTED** [review]
There is now duplicated code in contact.tsx. Please refactor to remove duplicated code.

**Comment by dakota002** [comment]
> There is now duplicated code in contact.tsx. Please refactor to remove duplicated code.

I addressed it in the most recent commit

**Review by lpsinger — APPROVED** [review]

**Pull request merged**